components/encryption/src/manager/mod.rs

```diff
@@ -337,15 +344,32 @@ impl Dicts {
         let duration = now.duration_since(UNIX_EPOCH).unwrap();
         let creation_time = duration.as_secs();
 
-        let (key_id, key) = generate_data_key(method);
-        let data_key = DataKey {
-            key,
-            method,
-            creation_time,
-            was_exposed: false,
-            ..Default::default()
-        };
-        self.rotate_key(key_id, data_key, master_key)
+        // Generate new data key.
+        for _ in 0..GENERATE_DATA_KEY_LIMIT {
+            let (key_id, key) = generate_data_key(method);
```

Comment:
Will there any side affects if generate too many keys (like rate limit or reach key capacity)?

Reply:
It is just a random generator, no side effect. On the other hand, the accepted keys will populate the key dict and never be GC-ed. In practice it's probably fine because rotate_period is long enough (7d).